feat: Implement category delete API

zerone1202 opened the pull request:
## #️⃣ Issue Number<!--- ex) #이슈번호, #이슈번호 -->

- Closes #59
- 관리자가 카테고리를 삭제할 수 있는 Soft Delete 방식의 API를 구현

## 📝 요약(Summary)<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
- DELETE /api/v1/admin/categories/{categoryId} 엔드포인트 추가
- CategoryController / Impl에 삭제 메서드 구현
- CategoryCommandService / Impl에 soft delete 로직 추가
- Category 엔티티의 delete() 메서드 활용하여 deletedAt 값 설정
- 일반 조회(findAllByDeletedAtIsNull, findByIdAndDeletedAtIsNull)에서 삭제된 카테고리는 자동 제외
- 관리자 권한(@PreAuthorize("hasRole('ADMIN'))) 적용
- 삭제 성공 메시지 `CATEGORY_DELETED` 추가

## 🛠️ PR 유형
어떤 변경 사항이 있나요?
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## ✅ PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.
- [x] 커밋 메시지 컨벤션에 맞게 작성

zerone1202 committed:
feat(category): Add category delete endpoint in controller

- DELETE /api/v1/admin/categories/{categoryId} 추가
- 관리자 권한 @PreAuthorize 적용
- 삭제 성공 시 공통 응답 반환

zerone1202 committed:
feat(category): Add deleteCategory method in controller interface

- 카테고리 삭제 API 인터페이스 메서드 추가
- DELETE /admin/categories/{categoryId} 명세 반영

zerone1202 committed:
feat(category): Add deleteCategory method in command service interface

- 삭제 서비스 동작 정의

zerone1202 committed:
feat(category): Implement category delete service with soft-delete

- findByIdAndDeletedAtIsNull로 유효한 카테고리 조회
- softDelete() 호출하여 deletedAt 기록
- 일반 조회에서 제외되도록 구조 유지

zerone1202 committed:
feat(category): Add softDelete method in Category entity

- deletedAt 값 업데이트하는 soft delete 기능 추가

zerone1202 committed:
feat(category): Add CATEGORY_DELETED success message

- 카테고리 삭제 성공 메시지 상수 추가

zerone1202 commented on src/main/java/com/golfRental/domain/category/service/command/CategoryCommandServiceImpl.java:
```diff
@@ -58,4 +58,13 @@ public CategoryUpdateResponse updateCategory(Long categoryId, CategoryUpdateRequ
                 .build();
     }
+
+    @Override
+    public void deleteCategory(Long categoryId) {
+
+        Category category = categoryRepository.findByIdAndDeletedAtIsNull(categoryId)
+                .orElseThrow(() -> new CategoryException(CategoryErrorCode.CATEGORY_NOT_FOUND));
+
+        category.softDelete();
+    }
```
해당 검증(post 사용 여부 체크)은 카테고리 도메인에서 해결할 수 있는 범위를 넘어 post 도메인의 비즈니스 로직을 추가로 다뤄야 합니다.  
제가 맡고 있는 도메인 스코프 밖의 변경이 포함되기 때문에 이번 PR에서는 적용하지 않겠습니다.

AllSungho commented on src/main/java/com/golfRental/domain/category/controller/CategoryControllerImpl.java:
```diff
@@ -79,4 +79,18 @@ public ResponseEntity<CommonApiResponse<CategoryUpdateResponse>> updateCategory(
+    ) {
+        categoryCommandService.deleteCategory(categoryId);
+
+        return CommonApiResponse.success(
```
CommonApiResponse에 삭제 성공 응답을 위해 특별히 마련된 deleteSuccess 메서드가 있습니다. success(null, ...) 대신 이 메서드를 사용하면 코드의 가독성과 의도를 더 명확하게 표현할 수 있습니다.
참고하시면 될 것 같습니다!!

zerone1202 replied:
오 제가 CommonApiResponse를 제대로 참고하지 않은 것 같네요. 참고 후 반영하겠습니다. 감사합니다.

zerone1202 committed:
refactor(category): Replace delete success response with deleteSuccess()

- CommonApiResponse.success(null, ...) 대신 deleteSuccess() 사용하도록 변경
- 삭제 API의 의도가 더 명확하게 드러나도록 가독성 개선
- 일관된 응답 포맷을 유지하기 위한 리팩토링

AllSungho reviewed (approved):
수정사항 반영 확인했습니다.
수고하셨습니다!!!
